src/meteodatalab/grib_decoder.py

```diff
@@ -366,7 +366,7 @@ def to_grib(loc: dict[str, xr.DataArray]):
     for idx_slice in product(*idx.values()):
         loc = {dim: value for dim, value in zip(idx.keys(), idx_slice)}
         array = field.sel(loc).values
-        metadata = md.override(to_grib(loc))
+        metadata = md.override(to_grib(loc)) if to_grib(loc) else md
```

Comment:
uhmmm yes, indeed. I had a hacked version of this (since pressure levels are not supported),  where I hacked the to_grib function to ignore the coordinate "pressure", that 's why to_grib returns an empty dict. 
But I see since to_grib is an internal function that it should returns always a dict with all coordinates defined, so the problem was coming from the additional modifications I made, so this has not much general value.
So I will close this.